### PR TITLE
Feature/mat 5946

### PR DESCRIPTION
**Pull request by RohitKandimalla**
## MADiE PR

Jira Ticket: [MAT-5946](https://jira.cms.gov/browse/MAT-5946) and [MAT-5926](https://jira.cms.gov/browse/MAT-5926) 
(Optional) Related Tickets:

### Summary

- A single zip file can be uploaded, virus scanning is performed on every file in the zip.
- A Spinner is added if the zip file import takes longer time.
- Made sure state is reset between consecutive import as there is an internal state for Dialog.
- This PR includes the complete flow of importing a test case that has valid folder structure, and removes measureReport and updates the existing test cases.

- Added import test case dialog
image
- Added import button on test case list page and updated existing import button to Import from Bonnie 
image

image

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/testCaseLanding/common/import/TestCaseImportDialog.tsx [hunk 1]
```diff
@@ -42,149 +40,164 @@ const TestCaseImportDialog = ({ open, handleClose, onImport }) => {
...
-        return;
-      }
-      if (patientBundles && patientBundles.length > 0) {
+  const onDrop = useCallback(async (importedZip) => {
+    removeUploadedFile();
+    setUploadingFileSpinner(true);
+    const zip = new JSZip();
+    let isValidImport = true;
+    zip.loadAsync(importedZip[0]).then((content) => {
+      Object.keys(content.files).forEach((filename) => {
```
Comment (adongare) [inline src/components/testCaseLanding/common/import/TestCaseImportDialog.tsx]: is this kind of reading the file? if yes, I think we are reading the file before virus scanning

Comment (RohitKandimalla) [reply src/components/testCaseLanding/common/import/TestCaseImportDialog.tsx]: Yeah, using JSZip. It can help loop through all the files

Comment (adongare) [reply src/components/testCaseLanding/common/import/TestCaseImportDialog.tsx]: you can send the whole file as it is for scanning. Once you get a clean scan result, you can process the zip if you need individual files

File: src/api/useTestCaseServiceApi.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import useServiceConfig from "./useServiceConfig";
 import { ServiceConfig } from "./ServiceContext";
 import { HapiOperationOutcome, TestCase } from "@madie/madie-models";
 import { useOktaTokens } from "@madie/madie-util";
-import { ScanValidationDto } from "./models/ScanValidationDto";
+import { ObjectError, ScanValidationDto } from "./models/ScanValidationDto";
```
Comment (adongare) [inline src/api/useTestCaseServiceApi.ts]: this import doesn’t look like being used anywhere

File: src/components/testCaseLanding/common/import/TestCaseImportDialog.tsx (hunk 1 above)
Comment (RohitKandimalla) [reply src/components/testCaseLanding/common/import/TestCaseImportDialog.tsx]: Done

**Review by chubert-sb — APPROVED** [review]

**Pull request merged**